### ISSUE TITLE
Unexpected result from pipe! expected 'P', instead received 'C'

### ISSUE THREAD

**Issue by B14CK-SPID3R** [googleprojectzero/winafl#454]
Hi
Is there any update to address this issue?

```
OS: Windows 11 Pro 24H2 26100.2605
DR: DynamoRIO-Windows-11.0.20071
Visual Studio 2022 Developer Command Prompt v17.7.5
```

```
C:\Users\user\Desktop\Fuzzing\1\winafl\build64\bin\Release>afl-fuzz.exe -i D:\tmp\gdiplus_input -o D:\tmp\gdiplus_output -D "C:\Users\user\Desktop\Fuzzing\1\DynamoRIO\bin64" -t 2000000 -- -coverage_module gdiplus.dll -fuzz_iterations 5000 -target_module test_gdiplus.exe -target_offset 0x10E0 -nargs 2 -- "C:\Users\user\Desktop\Fuzzing\1\winafl\build64\bin\Release\test_gdiplus.exe" @@
WinAFL 1.17 by <ifratric@google.com>
Based on AFL 2.43b by <lcamtuf@google.com>
[+] You have 32 CPU cores with average utilization of 0%.
[+] Try parallel jobs - see afl_docs\parallel_fuzzing.txt.
[*] Checking CPU core loadout...
[+] Found a free CPU core, binding to #0.
[+] Process affinity is set to 1.
[*] Setting up output directories...
[+] Output directory exists but deemed OK to reuse.
[*] Deleting old session data...
[+] Output dir cleanup successful.
[*] Scanning 'D:\tmp\gdiplus_input'...
[+] No auto-generated dictionary tokens to reuse.
[*] Creating hard links for all input files...
[*] Attempting dry run with 'id_000000'...

[-] PROGRAM ABORT : Unexpected result from pipe! expected 'P', instead received 'C'

         Location : run_target(), C:\Users\user\Desktop\Fuzzing\1\winafl\afl-fuzz.c:2920
```
According to my analysis, It returns to windows version. I compiled WinAFL in Windows 10 with the latest version of DR. Then I moved WinAFL and DR to a Windows 10 machine. Surprisingly, it worked on Windows 10, but it still didn't work on the Windows 11 machine.

**Comment by ifratric** [comment]
I don't know what the issue is, but some things you can try is
 - Building on the same machine where you're doing testing
 - If you are not already, using the latest DynamoRIO version from https://github.com/DynamoRIO/dynamorio/releases (make sure you build and run against the same DR version)
 - Using TinyInst instrumentation instead of DynamoRIO, see https://github.com/googleprojectzero/winafl/blob/master/readme_tinyinst.md for details.

**Comment by ifratric** [comment]
If it's otherwise behaving normally, then those warnings can be ignored.

**Comment by ifratric** [comment]
I just checked and I don't get those warnings on either my Win10 or Win11 machine against gdiplus.dll and WindowsCodecs.dll. It makes me think that there's maybe some antivirus / antimalware on your machine interfering with instrumentation. This could be the reason for issues with DynamoRIO as well.

**Comment by B14CK-SPID3R** [comment]
> I just checked and I don't get those warnings on either my Win10 or Win11 machine against gdiplus.dll and WindowsCodecs.dll. It makes me think that there's maybe some antivirus / antimalware on your machine interfering with instrumentation. This could be the reason for issues with DynamoRIO as well.

I've disabled AV and all exploit mitigation mechanisms in Windows Defender, then tried again. 
I also tried running it in safe mode, but the result was the same. 
Are you using the latest build of Windows 11?

**Comment by ifratric** [comment]
I was running on 23H2, I believe.

**Comment by B14CK-SPID3R** [comment]
Could you please try it with 24H2 (26100.2894)? I think you can reproduce the error message this way.

**Comment by B14CK-SPID3R** [comment]
> I was running on 23H2, I believe.

Sorry, did you check it? could you reproduce the issue?

**Comment by ifratric** [comment]
No, I don't want to update my Win11 machine to 24H2 just yet. However, as I wrote before, "Executing relative jump otside the current module" TinyInst warning is not an issue on its own, in fact the latest TinyInst from https://github.com/googleprojectzero/TinyInst doesn't even print this warning anymore.

**Comment by B14CK-SPID3R** [comment]
Thank you @ifratric , I approve it. Many errors have disappeared, but I still wish I could use DynamoRIO :) 
I’d appreciate it if you could also look into that issue.

**Comment by ifratric** [comment]
I looked a bit into DR issue in https://github.com/googleprojectzero/winafl/issues/466#issuecomment-2801286392.
Unfortunately, the issue appears to be on the DR side and not WinAFL side so I can't do much about it - I maintain TinyInst which I am the author of, but I don't have the resources to also investigate and fix DR issues. You can try reporting it to https://github.com/DynamoRIO/dynamorio though.
